src/Algebra/Module/Properties/Semimodule.agda

```diff
@@ -0,0 +1,40 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of semimodules.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                using (CommutativeSemiring)
+open import Algebra.Module.Bundles using (Semimodule)
+open import Level                  using (Level)
+
+module Algebra.Module.Properties.Semimodule
+  {r ℓr m ℓm : Level}
+  {semiring  : CommutativeSemiring r ℓr}
+  (semimod   : Semimodule semiring m ℓm)
+  where
+
+open CommutativeSemiring semiring
+open Semimodule          semimod
+open import Relation.Nullary.Negation using (contraposition)
```

Comment:
This should live in the imports above the module declaration as it doesn't depend on the parameters?

Reply:
Oh, I thought your previous comment in this regard was referring only to `import ...` lines and _not_ to `open import ...` lines. Now, I'm very confused, because my general sense of the standard library code is that the only `open import ...` lines that get pulled _above_ the main module declaration are those necessary, to specify the types of the main module parameters. Is that not the case?

Reply:
From the style guide:

> All imports should be placed in a list at the top of the file immediately after the module declaration.
>
> The list of imports should be declared in alphabetical order.
>
>If the module takes parameters that require imports from other files, then those imports only may be placed above the module declaration, e.g.

```agda
open import Algebra using (Ring)

module Algebra.Properties.Ring {a l} (ring : Ring a l) where

      ... other imports
```

Reply:
You're completely right! I was getting confused, apologies. Thanks for the spot :smile: